src/mlpack/core/data/string_encoding_policies/tf_idf_encoding_policy.hpp

```diff
@@ -0,0 +1,174 @@
+  template<typename MatType>
+  static void Encode(MatType& output, size_t value, size_t row, size_t /*col*/)
+  {
+    // Important since Mapping starts from 1 whereas allowed column value is 0.
+    // std::cout<<"divide "<<(tokenCount[row][value - 1] / row_size[row])<<"\n";
+    // std::cout<<"output size "<<output.n_rows<<" idfdict value "<<value-1<<" is "<<idfdict[value-1]<<"\n";
+    // std::cout<<"multipy "<<(output.n_rows / idfdict[value-1])<<" "<<std::log10(output.n_rows / idfdict[value-1])<<"\n";
+    output(row, value-1) = (tokenCount[row][value - 1] / row_size[row]) * 
+        std::log10(output.n_rows / idfdict[value-1]);
+    // std::cout<<"output at "<<row<<" "<<value-1<<" "<<output(row, value-1)<<"\n";
+  }
+
+  /** 
+  * The function performs the TfIdf encoding algorithm i.e. it writes
+  * the encoded token to the ouput.
+  * Overload function to accepted vector<vector<size_t>> as output type.
+  *
+  * @param output Output matrix to store the encoded results (sp_mat or mat).
+  * @param value The encoded token.
+  * @param row The row number at which the encoding is performed.
+  * @param col The row token number at which the encoding is performed.
+  */
+  template<typename OutputType>
+  static void Encode(std::vector<std::vector<OutputType> >& output, size_t value,
+                     size_t row, size_t /*col*/)
+  {
+    // Important since Mapping starts from 1 whereas allowed column value is 0.
+    // std::cout<<"divide "<<(tokenCount[row][value - 1] / row_size[row])<<"\n";
+    // std::cout<<"output size "<<output.size()<<" idfdict value "<<value-1<<" is "<<idfdict[value-1]<<"\n";
+    // std::cout<<"multipy "<<(output.size() / idfdict[value-1])<<" "<<std::log10(output.size() / idfdict[value-1])<<"\n";
+    output[row][value-1] = (tokenCount[row][value - 1] / row_size[row]) * 
+        std::log10(output.size() / idfdict[value-1]);
+    // std::cout<<"output at std::vector"<<row<<" "<<value-1<<" "<<output[row][value-1]<<"\n";
+  }
+
+  /**
+   * Serialize the class to the given archive.
+   */
+  template<typename Archive>
+  void serialize(Archive& /* ar */, const unsigned int /* version */)
+  {
+    // Nothing to serialize.
+  }
+
+  /*
+  * The function is used to create the datastrcutre will be important to find
+  * out idfvalue of words, and then wrtiting the output based on their count.
+  *
+  * @param row The row number at which the encoding is performed.
+  * @param numToken The count of token parsed till now.
+  * @param value The encoded token.
+  */
+  static void PreprocessToken(size_t row, size_t /*numTokens*/,
+                       size_t value)
+  {
+    if(row>=tokenCount.size())
+    {
+      row_size.push_back(0);
+      tokenCount.push_back(std::unordered_map<size_t, double>());
+    }
+    tokenCount.back()[value-1]++;
+    if(tokenCount.back()[value-1]==1)
+      idfdict[value-1]++;
+    row_size.back()++;
+  }
+ private:
+  static std::vector<std::unordered_map<size_t, double>> tokenCount;
```

Comment:
The `TfIdf` policy shouldn't contain static variables.

Reply:
Yes, I removed it .